dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -265,7 +268,7 @@ internal open class RumViewScope(
         val networkInfo = CoreFeature.networkInfoProvider.getLatestNetworkInfo()
         val errorType = event.type ?: event.throwable?.javaClass?.canonicalName
         val errorEvent = ErrorEvent(
-            date = event.eventTime.timestamp,
+            date = event.eventTime.timestamp + timeProvider.getServerOffsetMillis(),
```

Comment:
This will place the error event correctly on view's span only after the `timeProvider.getServerOffsetMillis()` gives "stable" results. AFAIK, right after starting the SDK but before completing the NTP sync the `timeProvider` time will vary - it will be becoming more precise over time. That said, when correcting the event time with **current** `timeProvider's` time, the events temporal relationship will be skewed, no?

The way we correct child events (error / action / resource) time on iOS is by capturing the "current server offset" on view start and then using this "frozen" value to offset time in related error / action / resource. This way we eliminate the effect of pending NTP sync. Don't you think this might be also necessary in Android SDK?

Reply:
The way I know the Kronos API already persists the last value between 2 processes so it will use the last value from a previous process.

Reply:
In any case this `Error` event will be sent before any view is actually captured in the application, I am not sure how this will work in this case.

Reply:
I see 👍. But even if Kronos caches the last known value, IMO we can't assume that the `getServerOffsetMillis()` will be constant for the entire process lifespan (e.g. for what I see, [we re-sync it every 30 minutes](https://github.com/DataDog/dd-sdk-android/blob/a7b4947ae34ed97be8e16d13adeac3c606d03c08/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt#L217-L218)). IMO if we depend on the current value each time, we will skew events time or even lose some events in web UI if their time is "corrected" so it occurs before the view started.

Although I put this comment on error event, I think it's valid for actions and resources as well.

Reply:
Ok let me open a thread in Slack and discuss more about this because there are several questions there.